Lib/textwrap.py

```diff
@@ -291,6 +293,35 @@ def _wrap_chunks(self, chunks):
                 else:
                     break
 
+            # If the last chunk on this line is all whitespace,
+            # and drop_whitespace is on,
+            # see if substituting a one-space chunk here leaves enough
+            # room to add the next non-whitespace chunk to the end.
+            # This fixes bpo-32397, where running wrap() twice on
+            # a paragraph might not be stable--the second run might
+            # produce different results than the first.  (If the first
+            # wrap() turned "foo.  bar" into "foo.\nbar", the second
+            # wrap() might turn it back into "foo. bar".  In that
+            # scenario, wrap() wil now produce "foo. bar".)
+            #
+            # This is all complicated slightly by fix_sentence_endings,
+            # where the chunk we add back in might need to be two spaces
+            # instead of one.
+            if (chunks and self.drop_whitespace
```

Comment:
According to the doc, drop_whitespace means "whitespace at the beginning and ending of every line (after wrapping but before indenting) is dropped".

But when you do below 
new_len = cur_len - len(cur_line[-1]) + len(spacer) + len(chunks[-1])
doesn't that amount to dropping whitespace within the line?


I see what you're trying to solve here, but I think that the definition of drop_whitespace, if I understand it correctly, makes wrap inherently non-idempotent.